x/crosschain/keeper/msg_server_add_to_intx_tracker_test.go

```diff
@@ -0,0 +1,198 @@
+			Proof:     proof,
+			BlockHash: block.Hash().Hex(),
+			TxIndex:   txIndex,
+		})
+		require.ErrorIs(t, types.ErrTxBodyVerificationFail, err)
+		_, found := k.GetInTxTracker(ctx, chainID, tx.Hash().Hex())
+		require.False(t, found)
+	})
+
+	t.Run("fail to add proof based tracker with wrong proof", func(t *testing.T) {
+		k, ctx, _, zk := keepertest.CrosschainKeeper(t)
+		chainID := int64(5)
+		txIndex, block, header, headerRLP, _, tx, err := sample.Proof()
+		require.NoError(t, err)
+		setupVerificationParams(zk, ctx, txIndex, chainID, header, headerRLP, block)
+		msgServer := keeper.NewMsgServerImpl(*k)
+		_, err = msgServer.AddToInTxTracker(ctx, &types.MsgAddToInTxTracker{
+			Creator:   sample.AccAddress(),
+			ChainId:   chainID,
+			TxHash:    tx.Hash().Hex(),
+			CoinType:  common.CoinType_Zeta,
+			Proof:     common.NewEthereumProof(ethereum.NewProof()),
+			BlockHash: block.Hash().Hex(),
+			TxIndex:   txIndex,
+		})
+		require.ErrorIs(t, types.ErrProofVerificationFail, err)
+		_, found := k.GetInTxTracker(ctx, chainID, tx.Hash().Hex())
+		require.False(t, found)
+	})
+	t.Run("normal user submit without proof", func(t *testing.T) {
+		k, ctx, _, _ := keepertest.CrosschainKeeper(t)
+		tx_hash := "string"
+		chainID := int64(5)
+		msgServer := keeper.NewMsgServerImpl(*k)
+		_, err := msgServer.AddToInTxTracker(ctx, &types.MsgAddToInTxTracker{
+			Creator:   sample.AccAddress(),
+			ChainId:   chainID,
+			TxHash:    tx_hash,
+			CoinType:  common.CoinType_Zeta,
+			Proof:     nil,
+			BlockHash: "",
+			TxIndex:   0,
+		})
+		require.ErrorIs(t, types.ErrTxBodyVerificationFail, err)
+		_, found := k.GetInTxTracker(ctx, chainID, tx_hash)
+		require.False(t, found)
+	})
+	t.Run("admin add  tx tracker", func(t *testing.T) {
+		k, ctx, _, zk := keepertest.CrosschainKeeper(t)
+		admin := sample.AccAddress()
+		setAdminPolicies(ctx, zk, admin)
+		tx_hash := "string"
+		chainID := int64(5)
+		msgServer := keeper.NewMsgServerImpl(*k)
+		_, err := msgServer.AddToInTxTracker(ctx, &types.MsgAddToInTxTracker{
+			Creator:   admin,
+			ChainId:   chainID,
+			TxHash:    tx_hash,
+			CoinType:  common.CoinType_Zeta,
+			Proof:     nil,
+			BlockHash: "",
+			TxIndex:   0,
+		})
+		require.NoError(t, err)
+		_, found := k.GetInTxTracker(ctx, chainID, tx_hash)
+		require.True(t, found)
+	})
+	t.Run("admin submit fake tracker", func(t *testing.T) {
+		k, ctx, _, zk := keepertest.CrosschainKeeper(t)
+		admin := sample.AccAddress()
+		setAdminPolicies(ctx, zk, admin)
+		tx_hash := "string"
+		chainID := int64(5)
+		msgServer := keeper.NewMsgServerImpl(*k)
+		_, err := msgServer.AddToInTxTracker(ctx, &types.MsgAddToInTxTracker{
+			Creator:   admin,
+			ChainId:   chainID,
+			TxHash:    "Malicious TX HASH",
+			CoinType:  common.CoinType_Zeta,
+			Proof:     nil,
+			BlockHash: "",
+			TxIndex:   0,
+		})
+		require.NoError(t, err)
+		_, found := k.GetInTxTracker(ctx, chainID, "Malicious TX HASH")
+		require.True(t, found)
```

Comment:
To confirm, the plan for now is to allow admin submitting whatever tx hash (they believe true) for now?

Reply:
yes , this test will fail when we change that , that's why added this